docs/path-resolution.rst

```diff
@@ -0,0 +1,454 @@
+
+.. code-block:: solidity
+    :caption: lib/src/../contract.sol
+
+    import "./util/./util.sol";         // source unit name: lib/src/../util/util.sol
+    import "./util//util.sol";          // source unit name: lib/src/../util/util.sol
+    import "../util/../array/util.sol"; // source unit name: lib/src/array/util.sol
+    import "../.././../util.sol";       // source unit name: util.sol
+
+.. note::
+
+    The use of relative imports containing leading ``../`` segments is not recommended.
+    The same effect can be achieved in a more reliable way by using direct imports with
+    :ref:`base path <base-path>` and :ref:`import remapping <import-remapping>`.
+
+.. index:: ! base path, --base-path
+.. _base-path:
+
+Base Path
+=========
+
+Base path specifies the directory that the Host Filesystem Loader will load files from.
+It is simply prepended to a source unit name before the filesystem lookup is performed.
+
+By default base path is empty, which leaves the source unit name unchanged.
+When the source unit name is a relative path, this results in the file being looked up in the
+directory the compiler has been invoked from.
+It is also the only value that results in absolute paths in source unit names being actually
+interpreted as absolute paths on disk.
+
+If the base path itself is relative, it is also interpreted as relative to the current working
+directory of the compiler.
+
+.. index:: ! remapping; import, ! import; remapping, ! remapping; context, ! remapping; prefix, ! remapping; target
+.. _import-remapping:
+
+Import Remapping
+================
+
+Base path and relative imports on their own allow you to freely move your project around the
+filesystem but force you to keep all files within a single directory and its subdirectories.
+When using external libraries it is often desirable to keep their files in a separate location.
+To help with that, the compiler provides another mechanism: import remapping.
+
+Remapping allows you to have the compiler replace import path prefixes with something else.
+For example you can set up a remapping so that everything imported from the virtual directory
+``github.com/ethereum/dapp-bin/library`` would actually receive source unit names starting with
+``dapp-bin/library``.
+By setting base path to ``/project`` you could then have the compiler find them in
+``/project/dapp-bin/library``
+
+The remappings can depend on a context, which allows you to configure packages to import,
+e.g. different versions of a library of the same name.
+
+.. warning::
+
+    Information about used remappings is stored in contract metadata so modifying them will result
+    in a slightly different bytecode.
+    This means that if you move your project files to different locations and use remappings to
+    avoid having to adjust the source code, your project will compile but will no longer produce the
+    exact same bytecode as without the remappings.
```

Comment:
The bytecode is different, because it's embedded metadata-hash is different, right?

Reply:
Exactly. I have modified the text now to say that explicitly.